kbase-extension/static/kbase/js/widgets/narrative_core/upload/stagingAreaViewer.js

```diff
@@ -407,57 +417,121 @@ define([
                             );
                         } else {
                             $('.kb-dropzone').css('min-height', '200px');
                             $tr.next().detach();
-                            delete this.openFileInfo[fileName];
+                            delete stagingAreaViewer.openFileInfo[fileName];
+                        }
+                    });
+
+                    $('td:eq(2)', row).find('.kb-staging-table-body__name').tooltip({
+                        title: $('td:eq(2)', row).find('.kb-staging-table-body__name').text(),
+                        placement: 'top',
+                        delay: {
+                            show: Config.get('tooltip').showDelay,
+                            hide: Config.get('tooltip').hideDelay
                         }
                     });
 
                     $('td:eq(2)', row).find('button[data-decompress]')
                     .off('click')
                     .on('click', e => {
-                        var fileName = $(e.currentTarget).data().decompress;
-                        var myFile = getFileFromName(fileName);
+                        const decompressButton = $(e.currentTarget);
+                        const fileName = decompressButton.data().decompress;
+                        const myFile = getFileFromName(fileName);
 
-                        $(e.currentTarget).replaceWith($.jqElem('i').addClass('fa fa-spinner fa-spin'));
+                        decompressButton.replaceWith($.jqElem('i').addClass('fa fa-spinner fa-spin'));
 
-                        this.stagingServiceClient.decompress({
+                        stagingAreaViewer.stagingServiceClient.decompress({
                             path: myFile.name
                         })
-                            .then(() => this.updateView())
-                            .fail(xhr => {
-                                console.error("FAILED", xhr);
-                                alert(xhr.responseText);
-                            });
+                        .then(() => stagingAreaViewer.updateView())
+                        .fail(xhr => {
+                            console.error("FAILED", xhr);
+                            alert(xhr.responseText);
+                        });
+
+                    });
 
+                    $('td:eq(5)', row).find('select').select2({
+                        placeholder: 'Select a type',
+                        containerCssClass: 'kb-staging-table-body__import-dropdown'
+                    }).on('select2:select', function(e) {
+
+                        $('td:eq(5)', row).find('.select2-selection').addClass('kb-staging-table-body__import-type-selected');
+                        
+                        //make checkbox for that row enabled
+                        //also set the data type so that we have the reference later when importing
+                        $('td:eq(0)', row).find('.kb-staging-table-body__checkbox-input')
```

Comment:
Enable checkbox when a file type is selected, set the data type, enable select all checkbox